src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp

```diff
@@ -208,13 +247,19 @@ void OPT_InitialiserLesBornesDesVariablesDuProblemeLineaire(PROBLEME_HEBDO* Prob
                 Var = CorrespondanceVarNativesVarOptim
                         ->NumeroDeVariableCoutOrigineVersExtremiteDeLInterconnexion[Interco];
 
-                if (CoutDeTransport->IntercoGereeAvecLoopFlow == OUI_ANTARES)
-                    Xmax[Var] = ValeursDeNTC->ValeurDeNTCOrigineVersExtremite[Interco]
-                                - ValeursDeNTC->ValeurDeLoopFlowOrigineVersExtremite[Interco];
+                if (SetToZeroLinkNTCForAdequacyPatchFirstStep == Data::setToZero
+                    || SetToZeroLinkNTCForAdequacyPatchFirstStep == Data::setStartEndToZero)
+                    Xmax[Var] = 0.;
                 else
-                    Xmax[Var] = ValeursDeNTC->ValeurDeNTCOrigineVersExtremite[Interco];
+                {
+                    if (CoutDeTransport->IntercoGereeAvecLoopFlow == OUI_ANTARES)
+                        Xmax[Var] = ValeursDeNTC->ValeurDeNTCOrigineVersExtremite[Interco]
+                                    - ValeursDeNTC->ValeurDeLoopFlowOrigineVersExtremite[Interco];
+                    else
+                        Xmax[Var] = ValeursDeNTC->ValeurDeNTCOrigineVersExtremite[Interco];
 
-                Xmax[Var] += 0.01;
+                    Xmax[Var] += 0.01;
+                }
```

Comment:
This is about transportation cost, I don't think we need to be touching that ?

Reply:
Yes. I tested and debugged this part. 
It is only activated if Hurdle costs are used and Var is indeed the transportation cost.
I though just to be sure :). But you are most probably right, it is not necessary.  
Ejected.